src/components/InventoryGroups/Modals/CreateGroupModal.js

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import validatorTypes from '@data-driven-forms/react-form-renderer/validator-types';
+import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
+import Modal from './Modal';
+import { nameValidator } from '../helpers/validate';
+import apiWithToast from '../utils/apiWithToast';
+import { createGroup, addSystemsToGroup } from '../utils/api';
+import { useDispatch } from 'react-redux';
+
+const asyncGroupNameValidation = async (value = '') => {
+    // do not fire validation request for empty name
+    if (value.length === 0) {
+        return undefined;
+    }
+    //api call to check that the name is valid and unique
+    /* const resp = await validateGroupName(value);
+    if (resp.data.isValid) {
+    // async validator has to throw error, not return it
+        throw 'Group name already exists';
+    } */
+};
+
+const validatorMapper = {
+    groupName: () => asyncGroupNameValidation
+};
+
+const createGroupSchema = {
```

Comment:
How about extracting modal assets into a new file?

Reply:
Good suggestion. will do that :)

Reply:
Moved the assets into the separate folder and file